src/main/java/com/researchspace/dao/hibernate/FullTextSearcherImpl.java

```diff
@@ -166,6 +191,19 @@ private List<BaseRecord> filterAndSortQueryList(
     resultList.addAll(hibernateBaseRecordList);
     resultList.addAll(attachmentBaseRecordList);
 
+    if (srchConfig.getFilters().isSharedFilter()) {
+      List<BaseRecord> sharedRecords =
+          recordGroupSharingDao.getSharedRecordsWithUser(srchConfig.getAuthenticatedUser());
+      resultList.retainAll(sharedRecords);
+    }
+
+    if (srchConfig.getFilters().isFavoritesFilter()) {
+      List<BaseRecord> userFavourites =
+          recordUserFavoritesDao.getFavoriteRecordsByUser(
+              srchConfig.getAuthenticatedUser().getId());
+      resultList.retainAll(userFavourites);
+    }
+
```

Comment:
This is to re-apply filtering that was previously handled by SearchManager.searchWorkspaceRecords(), right? Looking through the filters that used to be handled there, one missing here is ontology filter. You can actually see that filtering by ontology files no longer works after UI search on the branch (i.e. after applying 'ontology files' filter the result list shows all documents, not just ontologies).

[Workspace-ResearchSpace.webm](https://github.com/user-attachments/assets/6fb8700a-5360-4d2d-ad79-ecc0c61ccc35)